antiplag/views.py

```diff
@@ -3,15 +3,28 @@
 from rest_framework import status
 from rest_framework.parsers import MultiPartParser, FormParser
 from rest_framework.views import APIView
+from django.conf import settings
+from django.utils.translation import ugettext as _
+from django.views.decorators.cache import cache_page
+from django.utils.decorators import method_decorator
 
 from . import serializers
 from .models import Submission, Document
 from .constants import TEXT_SUBMISSION_NAME, CONTENT_TYPE_FILE, CONTENT_TYPE_TEXT
 from .tasks import process_documents
+from nlp.elastic import Elastic
 
-from django.conf import settings
-from nlp.text_preprocessing import preprocess_text
-from django.utils.translation import ugettext as _
+
+class Stats(APIView):
+    @method_decorator(cache_page(60 * 60), name="dispatch")
```

Comment:
I don't think you used `method_decorator` correctly. Either decorate class and pass function name (of function you want to decorate) to `name` arg or keep it as is and remove `name` arg.

Reply:
https://docs.djangoproject.com/en/3.1/topics/class-based-views/intro/#decorating-the-class

Reply:
yeah, seems like it I got confused and combined two ways. Not a big fan of decorators, to be honest - seems like a syntactic sugar and black box to me.

Mostly now, when developing docker, my PyCharm setup doesn't recognize the packages and I can't explore the functions from python using IntelliSense.